xknxproject/loader/project_loader.py

```diff
@@ -172,12 +173,37 @@ def _create_device(
 
         return device
 
+    @staticmethod
+    def __get_links_from_ets4(com_object: ElementTree.Element) -> str | None:
+        # Check if "Connectors" is available. This will always fail for ETS5/6
+        if (connectors := com_object.find("{*}Connectors")) is None:
+            return None
+
+        # Send GA is the primary GA, put it first in the list
+        if (send_ga := connectors.find("{*}Send")) is None:
+            return None
```

Comment:
A `Send` element should always be present if `Connectors` is there. Are you OK with an assertion here?

```suggestion
        send_ga = connectors.find("{*}Send")
        assert send_ga is not None, "'Send' element not present in 'Connectors' element"
```

Reply:
I think assertions shouldn't go into production code if we are not 100% sure they don't fire (we can't really since this is processing "user input" we don't control). We can just define an Exception (`UnexpectedProjectSchema` or something like that).
```suggestion
        if (send_ga := connectors.find("{*}Send")) is None:
            raise UnexpectedProjectSchema("'Send' element not present in 'Connectors' element")
```
or a `try: except:` block when you use it to raise.

Reply:
Well, this was more to make the code coverage check happy :wink: I'll rework the code a bit.

Reply:
Oh tbh I'm fine with raising an exception for such a thing is not covered. 
In fact we should use this pattern more often to just bail out when we encounter invalid data - this makes for far better readable error messages.

Reply:
Sure, failing fast is better. Still, here, even if we do not totally control the "user input" (the XML file), it is machine-generated and I doubt a `Connectors` element will show up without a child `Send` element. Checking for every unexpected scenario in this kind of parsing would probably make the code pretty cumbersome.

Reply:
Please, have a look at my latest changes and tell me if you like it or not.